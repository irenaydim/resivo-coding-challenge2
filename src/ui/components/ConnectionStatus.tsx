import { ConnectionStatus as ConnStatus } from "@/models/ConnectionStatus"
import { Typography } from "@mui/material"

interface ConnectionStatusProps {
  status: ConnStatus;
}

const ConnectionStatus = ({ status }: ConnectionStatusProps) => {
  return <Typography color={status === 'online' ? "success.main" : "error.main"}>{status}</Typography>
}

export default ConnectionStatus
