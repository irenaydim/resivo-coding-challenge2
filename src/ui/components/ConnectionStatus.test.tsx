import { render } from '@testing-library/react';
import { ConnectionStatus as ConnStatus } from '@/models/ConnectionStatus';
import ConnectionStatus from './ConnectionStatus';

const connectionStatus: ConnStatus = 'offline';

describe('ConnectionStatus', () => {
  it('should render correctly', () => {
    const { container } = render(<ConnectionStatus status={connectionStatus} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
