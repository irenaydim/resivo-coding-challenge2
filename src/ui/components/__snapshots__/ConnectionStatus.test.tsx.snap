// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectionStatus should render correctly 1`] = `
<p
  class="MuiTypography-root MuiTypography-body1 css-1nkhxq8-MuiTypography-root"
>
  offline
</p>
`;
